src/main/gpiiConnector.js

```diff
@@ -754,79 +738,53 @@ fluid.defaults("gpii.app.dev.gpiiConnector.qss", {
-        // Note that settings that doesn't have specific values such as "App / Text Zoom" will not receive setting updates
-        var qssSettingControls = fluid.filterKeys(channelSettingControls, fluid.keys(defaultQssSettingValues));
-
-        // add missing setting values if needed
-        qssSettingControls = gpii.app.dev.gpiiConnector.qss.applySettingDefaults(that, that.options.defaultQssSettingValues, qssSettingControls, value);
-
-        value.qssSettingControls = qssSettingControls;
-    }
-
-    return message;
-};
-
 
 /**
  * Fires the `onQssSettingsUpdate` event if needed to notify the QSS about setting changes.
  * Note that in case the changes are coming from snapset or active set update, they are not undoable (they indicate a full QSS reset).
- * @param {Component} that - The instance of `gpii.app.dev.gpiiConnector` component
+ * @param {gpii.app.dev.gpiiConnector} that - The instance of `gpii.app.dev.gpiiConnector` component
+ * @param {gpii.app} app - The instance of `gpii.app` component
  * @param {Object} message - The PSP channel decorated (with QSS data) massage
  */
-gpii.app.dev.gpiiConnector.qss.distributeQssSettings = function (that, message) {
+gpii.app.dev.gpiiConnector.qss.distributeQssSettings = function (that, app, message) {
     var payload = message.payload || {},
         value = payload.value || {},
-        qssSettingControls = value.qssSettingControls || {};
+        preferences = value.preferences || {},
+        settingGroups = value.settingGroups || [],
+        updateDefaultValues = false;
 
-    if (gpii.app.gpiiConnector.isPrefSetUpdate(payload)) {
-        fluid.log("gpiiConnector.qss Controls to be sent: ", value.qssSettingControls);
+    if (gpii.app.gpiiConnector.isPrefSetUpdate(payload) && fluid.isValue(settingGroups[0])) {
+        var settingControls = settingGroups[0].settingControls;
 
-        that.events.onQssSettingsUpdate.fire(
-            fluid.hashToArray(qssSettingControls, "path"), // set to the expected format
-            gpii.app.gpiiConnector.isFullPrefSetUpdate(that.previousState, value)
-        );
+        // update settings values with the default values from the PSP Channel when Morphic start or reset
+        if (!fluid.isValue(preferences.contexts)) {
+            fluid.each(settingControls, function (setting) {
+                if (fluid.isValue(setting.schema["default"])) {
+                    setting.value = setting.schema["default"];
+                }
+            });
+            // we need to overwrite the default values ONLY if we don't have preferences.contexts
+            // this is the occasion when there is no data (ie on launch, on reset, or keyIn)
+            updateDefaultValues = true;
+        }
+
+        fluid.log("gpiiConnector.qss Controls to be sent: ", settingControls);
+
+        // updating the whole settings group with the data from payload.value.settingGroups[0].settingControls
+        app.onPSPReadyForKeyIn.then(function () {
```

Comment:
Looks good - does it seem to work?

Reply:
Yup, its all working, but we a re having troubles with the tests. They seem to pass from time to time, like every 6th attempt or so. Looks like its related to the Survey delays. We will try to solve it by today, and will talk to you tomorrow about it.